rfcs/0136-stabilize-incrementally.md

```diff
@@ -0,0 +1,374 @@
+
+### Step 3: Attempt likewise splitting a store-and-language-only Nix
+
+For the same reason that a store-only Nix is useful for validating the store-only CLI, and ensuring it works with many *possible* higher layers, it is also useful to build a store-and-language-only Nix, without Flakes.
+Whether or not it is possible to actually do this is left to the Nix Team to decide, but it should be at least considered.
+
+### Step 4: Audit, refine, and stabilize the store-and-language-only Nix CLI
+
+This is the rest off the new CLI, without Flakes.
+Unlike the store-only Nix command which has yet to be implemented, this is easy to visualize today by enabling the `nix-command` feature without the `flakes` feature.
+This is a chance to discuss topics like:
+
+- Is `--file` good, or should we be able to specify something like `file#attribute` to mix files and attributes? (Without using Flakes.)
+- ~~Should all outputs be selected if one writes `foo.dev`?~~ Since fixed.
+- How can `nix repl` have a more normal CLI?
+
+Following the **gravity** principle, we will eventually want to pure eval and eval caching to be possible and easy to use without flakes; they should become part of this layer.
+The stabilized CLI must "make room" for these features becoming part of this layer.
+
+### Step 5: Audit, refine, and stabilize Flakes itself
+
+Finally, with the other less controversial interfaces stabilized, we can tackle the Flakes itself, the one remainder.
+This will require future RFCs.
+
+As stated in the previous step, following the **gravity**, pure eval and eval caching should be possible and easy to use without flakes.
+Ideally, by the time we get to this step, that is accomplished.
+If it isn't, we should at least make sure that it will be possible to do so later.
+It is OK to stabilize features that violate the layering principles, *only* so long as their stability does not impede fixing those violations later.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Having laid out the plan, let us now return to how the current situation is characterized and see if the various facts that the factions orient themselves are respected.
+
+## Layering principles
+
+Rammifications to the user experience from layering being public:
+
+  - The user has the option (it need not be mandatory!) to be aware of the layering and use it to learn Nix
+
+  - Both perspectives are equally valid, and neither is prioritized over the other
+
+  - While not going so far as to *insist* users are aware of layering and care about it, the layered archicture of Nix should be exposed to anyone that cares, and it shouldn't suddenly dissapear (as a mere implementation detail might).
+
+Some examples of ways the principles are upheld:
+
+- It is possible to use store without Nix lang
+  - (**publicity**, **compositionality**)
+  - Need `nix add-derivation` or similar to make this true!
```

Comment:
This has been resolved! In Nix 2.15, `nix derivation add` was added! See [2.15 Release Notes](https://nixos.org/manual/nix/unstable/release-notes/rl-2.15.html)

Reply:
Yes, I've been chipping away at the todo list here :). I am not sure that means we should remove them at out of date --- I sort of like examples of refinements of individual modules successfully performed, to provide evidence that we can efficiently "refine and clean up layers".

(The implicit contrast is would be that often deciding between options for future changes to Flakes/Fetchers is much harder because the design space is so wide open, there are not clear answers (Platonism cannot help us) and it's easier to get stuck.)